### PR TITLE
✨  add mutect2 module

### PR DESCRIPTION
**Pull request by dmiller15**
# Pull Request Template

## Description

Review everything. Adds the Mutect2 module CWL and scripts.

Part of https://github.com/d3b-center/bixu-tracker/issues/1337

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] https://cavatica.sbgenomics.com/u/d3b-bixu/dev-canine-workflow/tasks/8d4b199e-9af2-47de-81c3-d7b88350afac/#

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by migbro — APPROVED** [review]
Some minor comments, but looks good over all

File: tools/bcftools_filter_index.cwl [hunk 1]
```diff
@@ -0,0 +1,113 @@
+cwlVersion: v1.2
+class: CommandLineTool
+id: bcftools_reheader_sort
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/bcftools_filter_index.cwl:3]: I think I clean this up elsewhere.

[the remark this answers is not in the document]
Comment (dmiller15) [reply tools/bcftools_filter_index.cwl:3]: But I can still do it here then deal with the conflicts down the line.

File: scripts/calling_intervals_yaml_to_beds.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import sys
+import yaml
+
+inyaml = sys.argv[-1]
+payload = inyaml.split('/')[-1].split('.')[0]
```
[the remark this answers is not in the document]
Comment (dmiller15) [reply scripts/calling_intervals_yaml_to_beds.py:5]: This line is just getting a nameroot from the file path. Basically, the filename its like if the filename is `contigs.yaml`. The contents of the file will be:
```
contigs:
  - 1
  - 2
  - 3
  - 4
```

The reason I grab it from the fine name as opposed to hard code is that this script works with multiple inputs with different key names. I suppose I could make the assumption that there's only one key per file and just ask for it and store it. Might be an improvement.

File: tools/calling_intervals_yaml_to_beds.cwl [hunk 3]
```diff
@@ -0,0 +1,34 @@
+cwlVersion: v1.2
+class: CommandLineTool
+id: calling_intervals_yaml_to_beds
+doc: "Convert Calling Intervals YAML to interval BEDs"
+requirements:
+  - class: InlineJavascriptRequirement
+  - class: ShellCommandRequirement
+  - class: DockerRequirement
+    dockerPull: 'dmiller15/python-util:3.9'
```
Comment (sakshamphul) [inline tools/calling_intervals_yaml_to_beds.cwl:9]: Is this a public docker ?

**Review by sakshamphul — APPROVED** [review]

File: tools/calling_intervals_yaml_to_beds.cwl (hunk 3 above)
Comment (dmiller15) [reply tools/calling_intervals_yaml_to_beds.cwl:9]: It is public, yes. I will be moving this over to cavatica when I have the chance.

**Pull request merged**